cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -592,9 +596,16 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
                                 .build();
 
                         CertificateAuthority clusterCaConfig = kafkaAssembly.getSpec().getClusterCa();
-
                         // When we are not supposed to generate the CA but it does not exist, we should just throw an error
                         checkCustomCaSecret(clusterCaConfig, clusterCaCertSecret, clusterCaKeySecret, "Cluster CA");
+                        KafkaClusterTemplate kafkaClusterTemplate = kafkaAssembly.getSpec().getKafka().getTemplate();
+                        MetadataTemplate clusterCaMetadata = null;
+                        if (kafkaClusterTemplate != null) {
+                            ResourceTemplate clusterCaTemplate = kafkaClusterTemplate.getClusterCaCert();
+                            if (clusterCaTemplate != null) {
+                                clusterCaMetadata = clusterCaTemplate.getMetadata();
+                            }
+                        }
```

Comment:
In `KafkaCluster` you extracted the labels and annotations into separate fields. But here you seem to do it again instead of using the fields you set there. So I guess one or the other can be removed?